apps/studio/state/role-impersonation-state.tsx

```diff
@@ -0,0 +1,52 @@
+import { PropsWithChildren, createContext, useCallback, useContext } from 'react'
```

Comment:
can be a ts file right? no need for tsx?

Reply:
It's got JSX in it so it does need to be .tsx unfortunately